nestkernel/module_manager.h

```diff
@@ -0,0 +1,133 @@
+/*
+ *  module_manager.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef MODULE_MANAGER_H
+#define MODULE_MANAGER_H
+
+// Generated includes:
+#include "config.h"
+
+// C++ includes:
+#include <map>
+#include <string>
+
+// Includes from nestkernel:
+#include "manager_interface.h"
+
+// Includes from sli:
+#include "dictutils.h"
+
+// DynamicLoaderModule defined only if libltdl is available
+#ifdef HAVE_LIBLTDL
+#include <ltdl.h>
```

Comment:
Does it make sense to process any `#includes` if there is no LTDL?
```suggestion
// DynamicLoaderModule defined only if libltdl is available
#ifdef HAVE_LIBLTDL

#define MODULE_MANAGER_H

// Generated includes:
#include "config.h"

// C++ includes:
#include <map>
#include <string>

// Includes from nestkernel:
#include "manager_interface.h"

// Includes from sli:
#include "dictutils.h"

#include <ltdl.h>
```

Reply:
We need `config.h` to get `HAVE_LIBLTDL`, `dictutils.h` and `manager_interface.h` for the definition of the empty class when LTDL is missing, and we need to define `MODULE_MANAGE_H` in any case to avoid double inclusion of the header. So the only difference would be `map` and `string`,  and there I think it keeps the code simpler as is. The reduction in compile time due to the reduced includes would very limited.